tools/points-service/api.go

```diff
@@ -0,0 +1,316 @@
+package main
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"log"
+	"log/slog"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+)
+
+// PointsAPI holds references needed to handle API requests
+type PointsAPI struct {
+	logger *slog.Logger
+	db     *sql.DB
+	ps     *PointsService
+}
+
+// NewPointsAPI creates a new PointsAPI instance
+func NewPointsAPI(logger *slog.Logger, db *sql.DB, ps *PointsService) *PointsAPI {
+	return &PointsAPI{logger: logger, db: db, ps: ps}
+}
+
+// StartAPIServer starts the HTTP server with all routes
+func (p *PointsAPI) StartAPIServer(ctx context.Context, addr string) error {
+	r := mux.NewRouter()
+
+	// Define routes
+	r.HandleFunc("/health", p.HealthCheck).Methods("GET")
+	r.HandleFunc("/last_block", p.GetLastBlock).Methods("GET")
+	r.HandleFunc("/{receiver_type}/{receiver_address}", p.GetPointsForAddress).Methods("GET")
+	r.HandleFunc("/stats", p.GetTotalPointsStats).Methods("GET")
+	r.HandleFunc("/all", p.GetAllPoints).Methods("GET")
+
+	srv := &http.Server{
+		Addr:    addr,
+		Handler: r,
+	}
+
+	// Graceful shutdown
+	go func() {
+		<-ctx.Done()
+		if err := srv.Shutdown(context.Background()); err != nil {
+			p.logger.Error("HTTP server shutdown error", "error", err)
+		}
+	}()
+
+	p.logger.Info("Starting External Points API", slog.String("addr", addr))
+	if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+		return err
+	}
+	return nil
+}
+
+// HealthCheck endpoint checks if the API is alive
+// GET /health
+func (p *PointsAPI) HealthCheck(w http.ResponseWriter, r *http.Request) {
```

Comment:
If you want to add a health check, ideally you should ensure the points routine is running, subscriber for events is running etc.